src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs

```diff
@@ -131,10 +131,63 @@ public PackageArchiveReader BuildPackage(PackageBuilder builder, string outputPa
                 PrintVerbose(outputPath, builder);
             }
 
+            if (_packArgs.InstallPackageToV3Feed)
+            {
+                WriteResolvedNuSpecToPackageOutputDirectory(outputPath, builder);
+                WriteSHA512PackageHash(outputPath, builder);
+            }
+
             _packArgs.Logger.LogMinimal(String.Format(CultureInfo.CurrentCulture, Strings.Log_PackageCommandSuccess, outputPath));
             return new PackageArchiveReader(outputPath);
         }
 
+        /// <summary>
+        /// Writes the resolved NuSpec file to the package output directory.
+        /// </summary>
+        /// <param name="packageOutputPath">The output package path</param>
+        /// <param name="packageBuilder">The package builder</param>
+        private void WriteResolvedNuSpecToPackageOutputDirectory(string packageOutputPath, PackageBuilder packageBuilder)
+        {
+            string packageOutputDirectory = Path.GetDirectoryName(packageOutputPath);
+            string resolvedNuSpecOutputPath = Path.Combine(packageOutputDirectory, packageBuilder.Id + PackagingConstants.ManifestExtension);
```

Comment:
could you use this to get outputPath: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs#L978 
modify the function to accept a parameter which decides whether or not to append version. you can use it for the sha512 file too

Reply:
Resolved.